src/backend/main/go.main/contact/vcard.go

```diff
@@ -0,0 +1,49 @@
+// Copyleft (ɔ) 2019 The Caliopen contributors.
+// Use of this source code is governed by a GNU AFFERO GENERAL PUBLIC
+// license (AGPL) that can be found in the LICENSE file.
+
+package contact
+
+import (
+	"bufio"
+	"bytes"
+	"github.com/emersion/go-vcard"
+	"io"
+	"strings"
+)
+
+func parseBuffer(buffer io.Reader) (vcard.Card, error) {
+	dec := vcard.NewDecoder(buffer)
+	for {
+		card, err := dec.Decode()
+		if err != nil {
+			return vcard.Card{}, err
+		}
+		return card, nil
+	}
+}
+
+// Parse .vcf .vcard file, returting list of Card objects
+func ParseVcardFile(file io.Reader) ([]vcard.Card, error) {
+	cards := make([]vcard.Card, 0, 5)
+
+	scanner := bufio.NewScanner(file)
+	scanner.Split(bufio.ScanLines)
```

Comment:
instead of scanning each line until `END:VCARD` with `bufio.ScanLines` split func, then rebuild each line again,  you could have implemented your own split func that split on `VCARD` delimiters.

Reply:
Yes you're right, my first experimentation around vcard decoding does not handle file with multiples entries. I did something not correctly. I'll simplify my implementation

Reply:
done in #984482dda